src/main/webapp/scripts/itinerary-overview.js

```diff
@@ -83,25 +75,90 @@ async function getTripData(){
         waypointsLength++;
     }
 
-    await getPlaceDetails(waypointIDs)
-    geocodePlaceId(hotel_ID, waypointIDs);
+    await getTravelTimes(hotel_ID,waypointIDs);
 }
 
+//<------ Map Section ------>
 /**
- * Takes the hotel PlaceID and the waypoints PlaceID Array from getTripData and works to convert them from placeID's to
+ * This function takes the waypointID's from getTripData and runs them through a directions API fetch to see which one
+ * of the waypoints is this furthest away from the hotel. This waypoint is then set is the destination(end) of the route
+ * Then the function calls the geocoding and place details functions.
+ */
+async function getTravelTimes(){
+    let longestDuration = 0;
+    let longestDurationWaypointID = "";
+
+    for(let waypointID of waypointIDs){
+
+        const directionsResponse = await fetch(
```

Comment:
Future improvement tip: "await fetch" is a blocking call, the code stops execution until the response is return from Google API. JS is single threaded, meaning the whole page is frozen until all of these fetch calls finished.

Whenever you see an network operation inside a for-loop, consider to do
```
for (...){
   promiseList.push(fetch(...))
}
```

and wait for all promise is resolved: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

In this way, the fetch doesn't block itself, so all API calls happens in parallel. It works especially well when you have too many waypoints.

Reply:
Ok sounds good thank you!